operational_analysis/methods/plant_analysis.py

```diff
@@ -89,12 +89,12 @@ def __init__(self, plant, reanal_products=["merra2", "ncep2", "erai","era5"], un
         self.uncertainty_nan_energy = np.float64(uncertainty_nan_energy)
         
         # Check that selected time resolution is allowed
-        if time_resolution not in ['M','D']:
-            raise ValueError("time_res has to either be M (monthly, default) or D (daily)")
+        if time_resolution not in ['M','D','H']:
+            raise ValueError("time_res has to either be M (monthly, default) or D (daily) or H (hourly)")
         self.time_resolution = time_resolution
-        self._resample_freq = {"M": 'MS', "D": 'D'}[self.time_resolution]
-        self._hours_in_res = {"M": 30*24, "D": 1*24}[self.time_resolution]
-        self._calendar_samples = {"M": 12, "D": 365.25}[self.time_resolution]
+        self._resample_freq = {"M": 'MS', "D": 'D', 'H': 'H'}[self.time_resolution]
+        self._hours_in_res = {"M": 30*24, "D": 1*24, "H": 1}[self.time_resolution]
+        self._calendar_samples = {"M": 12, "D": 365, "H": 365*24}[self.time_resolution]
```

Comment:
Why was daily changed from 365.25 to 365?

Reply:
Adding to Jordan's comment, it seems like this changes (slightly) how many days will be used when long-term sampling the reanalysis data with daily resolution (ignoring leap years). Is there a reason for this?

Reply:
I had changed this because of two reasons:
1) the QC on availability and curtailment (lines ~572) was not working well at hourly resolution with the 365.25 choice. Basically, that QC step would always fail for not-leap years.
2) using 365.25 was messing up some of the IAV calculations. Basically, since 365.25 is an approximate value, the length of the long-term reanalysis dataset was not exactly 100% accurate. At fine (mostly hourly) resolution we sometimes were ending up with a few extra hours, that were being treated as a full new year during the IAV calculation, and this was leading to an extra year with very low production (just a few hours of data), which led to very large IAV.
Setting self. _calendar_samples  to 365 fixes 1), and it worked well for 2) too, even though it is not a perfect solution, either, as we're losing a few timestamps in the creation of the long-term reanalysis dataset, which would now be N years minus a few hours.

I've thought more about this, and I think the following solution works better to build the long-term reanalysis dataset, as it exactly selects the last N years of data as wanted:
        long_term_reg_inputs = ws_df[ws_df.index[-1] + ws_df.index.freq - pd.offsets.DateOffset(years = self._run.num_years_windiness):]
With this new way to create the LT reanalysis dataset, we can then stick with the self. _calendar_samples at 365, as that will now only be used for the availability and curtailment QC step.

The one issue I have now is that the IAV still is not working as expected with the long-term solution proposed above. The issue is connected to this command:
gross_lt_annual = reg_inputs_lt['gross_lt'].resample('12MS').sum().values
Let's say that we have an 11-year period (at hourly resolution) created with the more accurate offset command above that starts at 2008-12-31 01:00:00, and ends at 2019-12-31 00:00:00. Now, this is exactly 11 yers, so the newly-proposed command above works well. However, the resample command above will actually create 12 years of gross_lt_annual, instead of the expected 11 years. This is because the resample command starts from December 1 in creating the annual values (so each year will go Dec 1 - Nov 30). So year 1 would lose data for 1-30 December 2008, and we'd get an additional 12th year with just data from 1-30 December 2019. 
Any suggestions on how to fix this issue? I have looked at other ways to use the resample command here but could not find an easy fix.

Reply:
Okay, I think I have now fixed the remaining issue, see lines 844-847. It's not very elegant, but it does its job. Happy to receive suggestions on how to make this chunk of code nicer.